cal/cal/static/js/app.js

```diff
@@ -1,111 +1,124 @@
-var analyticsApp = angular.module('analyticsApp', ['nvd3', 'ui.calendar']);
+/*jslint devel: true */
+/*jslint browser: true */
+/*jslint jquery: true */
+/*global d3 */
+/*global getCookie */
+/*global moment */
 
-analyticsApp.controller('LoggedInCtrl', function LoggedInController($scope) {
+var analyticsApp = window.angular.module('analyticsApp', ['analyticsApp.services', 'nvd3', 'ui.calendar']);
+
+analyticsApp.controller('LoggedInCtrl', function LoggedInController() {
 });
 
-function TagsCtrl($scope, $http) {
-  var tagUrl = '/v1/tags';
-  $scope.tags = [];
+function TagListCtrl($scope, $http, CalendarRangeService, TagService) {
 
-  // add all the tags
-  $http({method: 'GET', url: tagUrl + '.json' }).
-  success(function successCallback(data) {
-    for (var i = 0; i < data.results.length; i++) {
-      var tag = data.results[i];
-      $scope.tags.push({
-        id: tag.id,
-        label: tag.label,
-        keywords: tag.keywords,
-        hours: tag.hours
-      });
+  var _this = this;
```

Comment:
If you use `this` in a callback, it will refer to the callback instead of the controller, which is what we want. Setting a `_this` allows us to refer to the controller.